datadog/dogshell/wrap.py

```diff
@@ -219,25 +221,42 @@ def build_event_body(cmd, returncode, stdout, stderr, notifications):
         )
 
 
+def generate_warning_codes(option, opt, options_warning):
+    try:
+        # options_warning should be a comma separated string
+        warning_codes = [options_warning.split(",")]
+        return warning_codes
+    except ValueError:
+        raise optparse.OptionValueError("option %s: invalid warning codes value(s): %r" % (opt, options_warning))
+
+
+class WarningOption(optparse.Option):
+    TYPES = optparse.Option.TYPES + ("warning_codes",)
+    TYPE_CHECKER = copy(optparse.Option.TYPE_CHECKER)
+    TYPE_CHECKER["warning_codes"] = generate_warning_codes
+
+
 def parse_options(raw_args=None):
     '''
     Parse the raw command line options into an options object and the remaining command string
     '''
     parser = optparse.OptionParser(usage="%prog -n [event_name] -k [api_key] --submit_mode \
-[ all | errors ] [options] \"command\". \n\nNote that you need to enclose your command in \
+[ all | errors | warnings] [options] \"command\". \n\nNote that you need to enclose your command in \
 quotes to prevent python executing as soon as there is a space in your command. \n \nNOTICE: In \
 normal mode, the whole stderr is printed before stdout, in flush_live mode they will be mixed but \
 there is not guarantee that messages sent by the command on both stderr and stdout are printed in \
-the order they were sent.", version="%prog {0}".format(get_version()))
+the order they were sent.", version="%prog {0}".format(get_version()), option_class=WarningOption)
```

Comment:
I doubt you want to replace the whole option class for all options here, right?

Reply:
Yeah that's incorrect, need to revert to how it previously was.

Reply:
Actually, I just tested this and the `WarningOption` class should take the default options and just add the `warning_codes` one to the whole option parser, so it looks like this is required here. Otherwise it'll throw:

```
optparse.OptionError: option --warning_codes: invalid option type: 'warning_codes'
```

Reply:
Oh ok, that might be an `optparse` specific thing, damn. I guess that's fine then if that's the way you're supposed to add option. :)